sentry_sdk/integrations/qdrant/qdrant.py

```diff
@@ -0,0 +1,523 @@
+import json
+from contextlib import contextmanager
+from decimal import Decimal
+from functools import wraps
+from typing import TYPE_CHECKING
+from urllib.parse import urlparse
+
+from google.protobuf.descriptor import FieldDescriptor
+from httpx import Response, Request
+
+import grpc
+
+import sentry_sdk
+from sentry_sdk.consts import OP, SPANDATA
+from sentry_sdk.integrations.httpx import HttpxIntegration
+from sentry_sdk.integrations.qdrant.consts import (
+    _DISALLOWED_REST_FIELDS,
+    _DISALLOWED_PROTO_FIELDS,
+    _qdrant_trie,
+    _IDENTIFIER,
+)
+
+# Hack to get new Python features working in older versions
+# without introducing a hard dependency on `typing_extensions`
+# from: https://stackoverflow.com/a/71944042/300572
+# taken from sentry_sdk.integrations.grpc.__init__.py
+if TYPE_CHECKING:
+    from typing import (
+        Any,
+        Callable,
+        Awaitable,
+        ParamSpec,
+        Optional,
+        Sequence,
+        Generator,
+    )
+    from grpc import Channel, ClientCallDetails
+    from grpc.aio import UnaryUnaryCall
+    from grpc.aio import Channel as AsyncChannel
+    from google.protobuf.message import Message
+    from sentry_sdk.tracing import Span, Transaction, _SpanRecorder
+    from sentry_sdk.integrations.qdrant import QdrantIntegration
```

Comment:
Are these only being used in type comments? If not, they should be imported outside the `if TYPE_CHECKING` block